src/grammar.coffee

```diff
@@ -154,7 +154,11 @@ URL_VALIDATOR = /// ^
             )
         | git/ (
               refs
-            | refs / heads (/[^/]+)?
+            | refs / (
+                .+
+                | heads (/[^/]+)?
```

Comment:
Not sure. It can technically end with /. But don't know if it's something you want to support, it's basically the same anyway..Tried adding a long custom ref here:

https://api.github.com/repos/mokkabonna/test1/git/refs/foo/test2 will be fine with the pattern you suggested, but https://api.github.com/repos/mokkabonna/test1/git/refs/foo/test2/ will not be fine. But it is a valid url technically.

If you only enumerate pulls|heads|tags you will not support custom refs, they are not very common to use, but I guess it happens.